ca/ca.go

```diff
@@ -252,6 +267,19 @@ func NewCertificateAuthorityImpl(
 	return ca, nil
 }
 
+func ResetSignatureCountForTesting() {
+	signatureCount.Reset()
+}
+
+func SignatureCountForTesting(certType string) int {
+	ch := make(chan prometheus.Metric, 10)
+	signatureCount.With(prometheus.Labels{"purpose": certType}).Collect(ch)
+	m := <-ch
+	var iom io_prometheus_client.Metric
+	_ = m.Write(&iom)
+	return int(iom.Counter.GetValue())
+}
```

Comment:
This is different than how it is done in http_test.go, but the mechanism used in http_test.go is really intrusive, especially as more and more counters are added.

Besides this way and the http_test.go way, an alternative would be to make signatureCount public and then move these two functions to ca_test.go.